Automattic/jetpack
# Repo Gardening: add logic to get label suggestions from OpenAI

## Proposed changes:

This will be triggered as part of the existing "Triage issues" task, only when an issue is opened.

It will fetch the list of "Feature" and "Feature Group" labels available in the repo, and then send the issue contents to OpenAI alongside with that list of labels.
OpenAI will respond with a list of labels that could match the issue.
We then apply those labels to the issue.

> [!NOTE]
> In order to help us test this change with real life data, I added extra logic to this PR. The automated labeling will now only happen on issues labeled with the "[Experiment] Automated labeling" label. Only when that label is present will we ask OpenAI for suggestions. In addition, and again to facilitate testing, the bot will also comment on the issue to say what was suggested and why. I'm hoping it can help us iterate on the prompt faster.

**To do**

- [x] Figure out OpenAI authentication: PCYsg-Vp7-p2
- [x] Include issue title in info sent to OpenAI

### Other information:


## Jetpack product discussion

* Internal reference: pdqkMK-1kt-p2

## Does this pull request change what data or activity we track or use?

* No

## Testing instructions:

This can only be tested in a fork of this repo so far, where you will have to push commits from this branch to trunk. There, add a new `OPENAI_API_KEY` secret with the key from the secret store, secret id 12313.

* Open a new issue.
* Label it with "[Experiment] Automated labeling"
* Let the action run (you can check the Actions tab in your repo to follow progress).
* You should see something like this (labels added, comment posted, label indicated issue has been processed): https://github.com/jeherve/jetpack/issues/152
